lib/vector-core/buffers/src/disk_v2/reader.rs

```diff
@@ -670,8 +778,21 @@ where
             // corrupted records, but hadn't yet had a "good" record that we could read, since the
             // "we skipped records due to corruption" logic requires performing valid read to
             // detect, and calculate a valid delta from.
-            if self.ledger.is_writer_done() && self.ledger.get_total_buffer_size() == 0 {
-                return Ok(None);
+            //
+            // TODO: Validate this with a test that messes with a data file without having to reload
+            // the buffer, since reloading the buffer will recalculate our buffer size.
```

Comment:
Intended to be addressed in this PR or later?

Reply:
Ooh, yes, addressed later.  I've been judiciously adding TODOs to the code but if I did them all in the same PR I had the thought to write a TODO for them.... 😅

Reply:
Opened #10898 for this one specifically.